src/main/java/com/google/sps/utility/ServletUtility.java

```diff
@@ -48,10 +50,10 @@ public static Cookie getCookie(HttpServletRequest request, String cookieName) {
     // This is unexpected - duplicate cookies are usually blocked by the browser (especially
     // when they are user set).
     if (cookies.size() > 1) {
-      System.out.printf("%d duplicate cookies with name %s\n", cookies.size(), cookieName);
-      cookies.forEach(c -> System.out.println(c.getValue()));
-
-      return null;
+      String errorMessage =
+          String.format("%d duplicate cookies with name %s\n", cookies.size(), cookieName);
+      errorMessage += cookies.stream().map(Cookie::getValue).reduce((a, b) -> a + "\n" + b);
+      throw new IOException(errorMessage);
```

Comment:
Is `IOException` really the best type of exception to use? This has nothing to do with disk or network operations.

Reply:
Fair enough - I created a custom error type for this